synapse/third_party_rules/access_rules.py

```diff
@@ -441,6 +465,21 @@ def _get_rule_from_state(state_events):
             rule = access_rules.content.get("rule")
         return rule
 
+    @staticmethod
+    def _get_join_rule_from_state(state_events):
+        """Extract the room's join rule from the given set of state events.
+
+        Args:
+            state_events (dict[tuple[event type, state key], EventBase]): The set of state
+                events.
+        Returns:
+            str, the name of the join rule (either "public", or "invite")
+        """
+        join_rule_event = state_events.get((EventTypes.JoinRules, ""))
+        if join_rule_event is None:
+            return ""
```

Comment:
Huh? Shouldn't this be the default join rule or `None`?

Reply:
Duh, I apparently have troubles switching between a strongly typed language and Python ._.

Reply:
Proper strongly typed languages would use an optional type :p